lib/deref.js

```diff
@@ -181,8 +187,14 @@ module.exports = {
           resolveTo, stack, _.cloneDeep(seenRef), stackLimit);
       }
       return { oneOf: _.map(schema.oneOf, (schemaElement) => {
-        return this.resolveRefs(schemaElement, parameterSourceOption, components, schemaResolutionCache, resolveFor,
-          resolveTo, stack, _.cloneDeep(seenRef), stackLimit);
+        PROPERTIES_TO_ASSIGN_ON_CASCADE.forEach((prop) => {
```

Comment:
@LuisTejedaS Will the same handling be needed for `allOf` as well?

Reply:
allOf seems to merge properties, so in this case the parent properties should be there, there is no need to copy them cc @erfemega